src/H5PB.c

```diff
@@ -1302,6 +1302,73 @@ H5PB_write(H5F_shared_t *f_sh, H5FD_mem_t type, haddr_t addr, size_t size, const
     FUNC_LEAVE_NOAPI(ret_value)
 } /* end H5PB_write() */
 
+/*-------------------------------------------------------------------------
+ * Function:    H5PB_enabled
+ *
+ * Purpose:     Check if the page buffer may be enabled for the specified
+ *              file and data access type.
+ *
+ * Return:      Non-negative on success/Negative on failure
+ *
+ * Programmer:  Neil Fortner
+ *
+ *-------------------------------------------------------------------------
+ */
+htri_t
+H5PB_enabled(H5F_shared_t *f_sh, H5FD_mem_t type)
+{
+    H5PB_t *page_buf;          /* Page buffering info for this file */
+    hbool_t bypass_pb = FALSE; /* Whether to bypass page buffering */
+    htri_t  ret_value;         /* Return value */
+
+    FUNC_ENTER_NOAPI(FAIL)
+
+    /* Sanity checks */
+    HDassert(f_sh);
+
+    /* Get pointer to page buffer info for this file */
+    page_buf = f_sh->page_buf;
+
+#ifdef H5_HAVE_PARALLEL
+    if (H5F_SHARED_HAS_FEATURE(f_sh, H5FD_FEAT_HAS_MPI)) {
+#if 1
+        bypass_pb = TRUE;
+#else
+        /* MSC - why this stopped working ? */
+        int mpi_size;
+
+        if ((mpi_size = H5F_shared_mpi_get_size(f_sh)) < 0)
+            HGOTO_ERROR(H5E_PAGEBUF, H5E_CANTGET, FAIL, "can't retrieve MPI communicator size")
+        if (1 != mpi_size)
+            bypass_pb = TRUE;
+#endif
```

Comment:
Should we remove the dead code?

Reply:
Hmm, does anyone know the answer to Mohamad's question? BTW this was just taken from H5PB_read/write for consistency (it needs to exactly match those functions' decisions on whether to use the page buffer).

Reply:
It's very likely this is something that has come and gone in terms of working or not. I use the same mpi_get_size routine in a check for parallel compression with a comm size of 1 and it works fine the last I checked, so it's likely this works fine as well; testing that theory would of course be a good idea.

Reply:
I suspect the issue isn't (wasn't?) that mpi_get_size isn't working, it's that enabling the page buffer with 1 rank doesn't work. I haven't tried it though.